controller/uninstall_controller.go

```diff
@@ -470,17 +485,19 @@ func (c *UninstallController) deleteInstanceManagers(instanceManagers map[string
 		err = errors.Wrapf(err, "Failed to delete instance managers")
 	}()
 	for _, im := range instanceManagers {
-		logFields := logrus.Fields{
-			"type": "instanceManager",
-			"name": im.Name,
-		}
+		log := c.logger.WithFields(
```

Comment:
Add a getLoggerForInstanceManager function to the instance-manager-controller, same idea as above for the engine.
The idea is to have a getLoggerX for each crd type.